ank-core/src/main/kotlin/io/kategory/ank/interpreter.kt

```diff
@@ -142,17 +151,42 @@ fun compileCodeImpl(snippets: Map<File, ListKW<Snippet>>, classpath: ListKW<Stri
                                     it.lang to seManager.getEngineByExtension(extensionMappings.getOrDefault(it.lang, "kts"))
                                 }
                                 .toMap()
+
+                if (codeBlocks.find { it.compile }.nonEmpty()) {
+                    val tmpFile = createTempFile(prefix = file.nameWithoutExtension + "__", suffix = ".kts")
+
+                    val (imports, code) = codeBlocks.fold(emptySet<String>() toT emptyList<String>()) { acc, snippet ->
+                        snippet.code.lines().fold(acc) { (imports, code), line ->
+                            if (line.startsWith("import")) imports + line else code + line
+                            imports toT code
+                        }
+                    }
+
+                    tmpFile.printWriter().use {
+                        it.println(imports.joinToString("\n"))
+                        it.println(code.joinToString("\n"))
+                    }
+
+                    val args = listOf("-Xcoroutines=enable", "-cp", classPath, "-d", file.parentFile.absolutePath, tmpFile.absolutePath).toTypedArray()
+
+                    //If compilation fails compiler shuts down VM. However correct output is shown in Gradle.
+                    K2JVMCompiler.main(args)
```

Comment:
what happens if the code snippet is `java:ank:compile` ?

Reply:
Need to check!